examples/systemcalls/systemcalls.c

```diff
@@ -1,46 +1,68 @@
+/*
+ * @File name (systemcalls.c)
+ * @File Description (Redited for AESD Assignment-3 using various systemcalls
+ * @Edited by AYSWARIYA KANNAN)
+ * @Date (2/04/2023)
+ * @Attributions :
+ */
 #include "systemcalls.h"
-
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 /**
  * @param cmd the command to execute with system()
  * @return true if the command in @param cmd was executed
  *   successfully using the system() call, false if an error occurred,
  *   either in invocation of the system() call, or if a non-zero return
  *   value was returned by the command issued in @param cmd.
-*/
+ */
 bool do_system(const char *cmd)
 {
 
-/*
- * TODO  add your code here
- *  Call the system() function with the command set in the cmd
- *   and return a boolean true if the system() call completed with success
- *   or false() if it returned a failure
-*/
+    /*
+     * TODO  add your code here
+     *  Call the system() function with the command set in the cmd
+     *   and return a boolean true if the system() call completed with success
+     *   or false() if it returned a failure
+     */
+    if (cmd == NULL) // check if pointer is NULL then returns false
+    {
+        return false;
+    }
+    int p = system(cmd); // call system() function
+    if (p == -1)         // if it returns -1 means call unsuccessful
+    {
+        return false;
```

Comment:
Check the requirements... I think you need additional checks here to meet the assignment requirements.  I don't think this is sufficient to know what the exit status was of the system() call.  I think you need WEXITSTATUS here.